internal/db/datatype_test.go

```diff
@@ -0,0 +1,125 @@
+package db
+
+import (
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestBool(t *testing.T) {
```

Comment:
these tests look good! 

I'd like to see you refactor them to be table based (see https://github.com/golang/go/wiki/TableDrivenTests)